### DISCUSSION TITLE
Media Files - Naming and Storing

### DISCUSSION THREAD

**Discussion by webtroter** [inventree/InvenTree#8294] [Ideas]
Hi,

After uploading my first files (part images and attachment), I saw that the uploaded file keeps the original file name (or adds a suffix if there is already a file with the same name).

I was thinking that it might be better to rename the uploaded files to an internal file name (just some ID or ID+relatedID).

Also, I think that having the ability to use a object-storage backend for the uploaded medias might be a useful feature.

Probably related in some way to https://github.com/inventree/InvenTree/issues/5703

**Comment by SchrodingersGat** [comment]
Is there any issues caused by the current naming scheme?

And you should be able to use any of django's supported backends for media files

**Reply by webtroter** [reply]
I'm thinking too much I think 😅😅

I have no problem. 

Thanks for taking the time to reply.

I will close this discussion, since even I am not sure how to contribute to it. 😅
